File: project-euler/2. Even Fibonacci Numbers/soln.py
def even_fibonacci_sum(limit):
    a, b = 1, 2
    total = 0
    while b <= limit:
        if b % 2 == 0:
            total += b
        a, b = b, a + b
    return total
print(even_fibonacci_sum(4000000))

File: project-euler/2. Even Fibonacci Numbers/soln.R
even_fibonacci_sum <- function(limit) {
  a <- 1
  b <- 2
  total <- 0
  while (b <= limit) {
    if (b %% 2 == 0) {
      total <- total + b
    }
    temp <- a + b
    a <- b
    b <- temp
  }
  
  return(total)
}
print(even_fibonacci_sum(4000000))

